### PR TITLE
[MRESOLVER-10] New class 'TransitiveDependencyManager'

### PR DESCRIPTION
**Pull request by slachiewicz**
TransitiveDependencyManager -  supporting transitive dependency management

Rebased, splitted and and style fixed from #9

### CODE REVIEW

File: maven-resolver-util/src/main/java/org/eclipse/aether/util/graph/manager/TransitiveDependencyManager.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+        for ( Dependency managedDependency : context.getManagedDependencies() )
+        {
+            Artifact artifact = managedDependency.getArtifact();
+            Object key = getKey( artifact );
+
+            String version = artifact.getVersion();
+            if ( version.length() > 0 && !versions.containsKey( key ) )
+            {
+                if ( versions == managedVersions )
```
Comment (michael-o) [inline maven-resolver-util/src/main/java/org/eclipse/aether/util/graph/manager/TransitiveDependencyManager.java]: Why does this use reference equality?

**Comment by michael-o** [comment]
Looks fine to me, but this isn't used by default, correct? It has to be used explicitly?

**Comment by slachiewicz** [comment]
Yes, correct- later we need to add changes in maven to use this feature.

File: maven-resolver-util/src/main/java/org/eclipse/aether/util/graph/manager/TransitiveDependencyManager.java (hunk 1 above)
Comment (slachiewicz) [reply maven-resolver-util/src/main/java/org/eclipse/aether/util/graph/manager/TransitiveDependencyManager.java]: that was in original PR

**Comment by jira-importer** [comment]
Resolve #1175

**Comment by jira-importer** [comment]
Resolve #1184

**Comment by jira-importer** [comment]
Resolve #1191

**Pull request closed without merging**